packages/playwright-core/src/server/browser.ts

```diff
@@ -90,6 +91,17 @@ export abstract class Browser extends SdkObject {
     return context;
   }
 
+  async newContextForReuse(params: channels.BrowserNewContextForReuseParams, metadata: CallMetadata): Promise<{ context: BrowserContext, needsReset: boolean }> {
+    const hash = BrowserContext.reusableContextHash(params);
+    if (!this._contextForReuse || hash !== this._contextForReuse.hash || !this._contextForReuse.context.canResetForReuse()) {
+      if (this._contextForReuse)
+        await this._contextForReuse.context.close(metadata);
+      this._contextForReuse = { context: await this.newContext(metadata, params), hash };
+      return { context: this._contextForReuse.context, needsReset: false };
```

Comment:
I don't see where would we do the following code in this `needsReset: false` case.

```js
    for (const key of paramsThatAllowContextReuse)
      (this._options as any)[key] = params[key];
```